aioredis/connection.py

```diff
@@ -39,6 +39,7 @@
     'PSUBSCRIBE', b'PSUBSCRIBE',
     'UNSUBSCRIBE', b'UNSUBSCRIBE',
     'PUNSUBSCRIBE', b'PUNSUBSCRIBE',
+    'PING', b'PING',
```

Comment:
Yes, it will fail.
You can extend condition of that `if` to something like:
```python
if self._in_pubsub and not (is_pubsub or is_ping):
    ...
```